### PR TITLE
Implement ``Allocatable`` in LLVM for different ASR types

### PR DESCRIPTION
**Pull request by czgdp1807**
In LLVM backend, now ``Allocatable`` and ``Pointer`` have the same physical type.

https://github.com/lfortran/lfortran/commit/5bbeff9ca315f94080eb17e772ebf68df3fb6df2
https://github.com/lfortran/lfortran/commit/8657f94a4aa96eca36a791307bf1c3c0c1c7fa37
https://github.com/lfortran/lfortran/commit/0a1f69295ca2e403043cbc0fbd342eef4edf61cb
https://github.com/lfortran/lfortran/commit/e853404cb08a709a50c192452b91cfd62fa1bc5b
https://github.com/lfortran/lfortran/commit/8efea0d5494c57109c6b0534ca1c58a780c6add3
https://github.com/lfortran/lfortran/commit/60bd50218f8e7c84eaac0dae2d2b947c829bbd74
https://github.com/lfortran/lfortran/commit/04a164aa1cbbd900d68c7aa7ec96deb077366458
https://github.com/lfortran/lfortran/commit/24535bd2a85299c929df6cfe693c67c48bd01d58
https://github.com/lfortran/lfortran/commit/9e3b9b69b1be04a589b8ba732c07939137245230
https://github.com/lfortran/lfortran/commit/042d790ab758795ff39faaa6a3277c666baae66b

### CODE REVIEW

**Comment by certik** [comment]
Great job on this!

**Comment by certik** [comment]
Looks good.

File: src/libasr/asr_verify.cpp [hunk 1]
```diff
@@ -1030,6 +1030,15 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
         visit_ttype(*x.m_type);
     }
 
+    void visit_Allocate(const Allocate_t &x) {
+        for( size_t i = 0; i < x.n_args; i++ ) {
+            require(ASR::is_a<ASR::Allocatable_t>(*ASRUtils::expr_type(x.m_args[i].m_a)) ||
+                    ASR::is_a<ASR::Pointer_t>(*ASRUtils::expr_type(x.m_args[i].m_a)),
+                "Allocate should only be called with  Allocatable or Pointer type inputs");
+        }
+        BaseWalkVisitor<VerifyVisitor>::visit_Allocate(x);
```
Comment (certik) [inline src/libasr/asr_verify.cpp:1039]: What does this do --- it calls the parent function, which by default visits all the arguments?

**Review by certik — APPROVED** [review]
I think this is fine. I reviewed it as much as I could. If something breaks, we'll fix it later. We need this change.

File: src/libasr/asr_verify.cpp (hunk 1 above)
Comment (czgdp1807) [reply src/libasr/asr_verify.cpp:1039]: Yes. Didn't want to duplicate the logic of parent function here, so just calling it. Also, logic of parent function might change in future (who knows?), duplication will require to make edits here as well.

**Comment by czgdp1807** [comment]
There will be some divergences from LPython regarding printing of pointers. That should be fixed by #1844 soon.

**Pull request merged**